packages/vite/src/node/utils.ts

```diff
@@ -769,22 +769,8 @@ export function processSrcSetSync(
   )
 }
 
-const cleanSrcSetRE =
-  /(?:url|image|gradient|cross-fade)\([^)]*\)|"([^"]|(?<=\\)")*"|'([^']|(?<=\\)')*'|data:\w+\/[\w.+\-]+;base64,[\w+/=]+/g
 function splitSrcSet(srcs: string) {
-  const parts: string[] = []
-  // There could be a ',' inside of url(data:...), linear-gradient(...), "data:..." or data:...
-  const cleanedSrcs = srcs.replace(cleanSrcSetRE, blankReplacer)
-  let startIndex = 0
-  let splitIndex: number
-  do {
-    splitIndex = cleanedSrcs.indexOf(',', startIndex)
-    parts.push(
-      srcs.slice(startIndex, splitIndex !== -1 ? splitIndex : undefined),
-    )
-    startIndex = splitIndex + 1
-  } while (splitIndex !== -1)
-  return parts
+  return srcs.split(/,\s+/)
```

Comment:
It seems the algorithm is a bit more complex, see:
> If present, its value must consist of one or more [image candidate strings](https://html.spec.whatwg.org/multipage/images.html#image-candidate-string), each separated from the next by a U+002C COMMA character (,). If an [image candidate string](https://html.spec.whatwg.org/multipage/images.html#image-candidate-string) contains no descriptors and no [ASCII whitespace](https://infra.spec.whatwg.org/#ascii-whitespace) after the URL, the following [image candidate string](https://html.spec.whatwg.org/multipage/images.html#image-candidate-string), if there is one, must begin with one or more [ASCII whitespace](https://infra.spec.whatwg.org/#ascii-whitespace).

From https://html.spec.whatwg.org/multipage/images.html#srcset-attribute

Reply:
I updated my PR with a different approach. I added an additional pattern in the `cleanSrSetRE` RegEx. This pattern matches a string that contains a question mark (not at the beginning or the end of the string) followed by a comma. This pattern is intended to match image URLs with query strings that contain commas. 

If matched, the string is replaced by blank spaces so the following code can split the srcset by commas that specifically follow descriptors.